Detectors/MUON/MID/Raw/include/MIDRaw/RawFileReader.h

```diff
@@ -0,0 +1,66 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file   MIDRaw/RawFileReader.h
+/// \brief  MID raw file reader
+/// \author Diego Stocco <Diego.Stocco at cern.ch>
+/// \date   25 November 2019
+#ifndef O2_MID_RAWFILEREADER_H
+#define O2_MID_RAWFILEREADER_H
+
+#include <fstream>
+#include <vector>
+#include <array>
+#include "MIDRaw/CrateParameters.h"
+#include "MIDRaw/RawBuffer.h"
+#include "MIDRaw/RawUnit.h"
+
+namespace o2
+{
+namespace mid
+{
+template <typename T>
+class RawFileReader
+{
+ public:
+  bool init(const char* inFilename, bool readContinuous = false);
+  bool readAllGBTs(bool reset = true);
+  bool readHB(bool sendCompleteHBs = true);
+  void clear();
+
+  /// Gets the number of HBs read
+  unsigned long int getNumberOfHBs(uint16_t feeId) { return mHBCounters[feeId]; }
```

Comment:
if this function time critical ? if not I would use std::array::at instead of operator[] to get bound checking

Reply:
Done